### PR TITLE
Output Shifter upgrade (naming and data structure)

### PR DESCRIPTION
**Pull request by GioCC**
## Description of changes

Addresses issues #137 (first commit) and #138 (second commit)

The naming upgrade only affects source code, no difference in functions or performance.

The data structure changes the fixed-length buffer with a byte array; there is a slight penalty in Flash space (+18 bytes), though - surprisingly - if we dispense with one single boundary check for an argument (in setPin()) there is even a slight gain (-12 bytes)!

_The second part still needs testing_

### CODE REVIEW

File: src/MF_Modules/MFOutputShifter.cpp [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    updateShiftRegister();
+    delay(50);
+  }
+  if(mode==0) {
+    _outputBuffer[_moduleCount-1] = 0;
+    updateShiftRegister();
+  }
+}
+
+void MFOutputShifter::test()
```
Comment (neilenns) [inline src/MF_Modules/MFOutputShifter.cpp:95]: This change is unrelated to the purpose of this PR and doesn't have a tracking issue. Did you mean to include it?

**Review by neilenns — CHANGES_REQUESTED** [review]

File: src/MF_Modules/MFOutputShifter.cpp (hunk 1 above)
Comment (GioCC) [reply src/MF_Modules/MFOutputShifter.cpp:95]: Not really. I was trying to introduce a maybe more useful, possibly less "expensive" test mode, when I discovered that the `test()` method is not invoked anywhere, so I forgot to follow up on that.

Comment (neilenns) [reply src/MF_Modules/MFOutputShifter.cpp:95]: Can you push an update that removes it then? Thanks!

**Comment by GioCC** [comment]
Done!

File: src/MF_Modules/MFOutputShifter.cpp [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+// Copyright (C) 2021
+
+#include "MFOutputShifter.h"
+
+MFOutputShifter::MFOutputShifter()
+{
+  _initialized = false;
+}
+
+void MFOutputShifter::setPin(uint8_t pin, uint8_t value, uint8_t refresh)
```
Comment (neilenns) [inline src/MF_Modules/MFOutputShifter.cpp:12]: What is this "refresh" thing? It doesn't get passed in anywhere that I can see and is unrelated to the naming changes.

**Review by neilenns — CHANGES_REQUESTED** [review]

File: src/MF_Modules/MFOutputShifter.cpp (hunk 2 above)
Comment (GioCC) [reply src/MF_Modules/MFOutputShifter.cpp:12]: It is used in SetPins(); it allows to change the output values deferring the transmission once all changes have been done. This way, SetPins() can use SetPin() (instead of duplicating the code) because it otherwise it would send all data every time.

Comment (neilenns) [reply src/MF_Modules/MFOutputShifter.cpp:12]: Thanks for the explanation! This change wasn't listed in the PR description so it wasn't clear if it was intentional as part of the rename work (since it's unrelated to the two mentioned issues).

**Review by neilenns — APPROVED** [review]

File: src/MF_Modules/MFOutputShifter.cpp (hunk 2 above)
Comment (GioCC) [reply src/MF_Modules/MFOutputShifter.cpp:12]: Actually I saw it more as part of the update related to the changes in the data structure, even if not strictly. It's internal management more than anything else, but I see it's clearer now. Thanks.

**Review by DocMoebiuz — APPROVED** [review]
Thanks!
